File: src/main/java/org/example/util/ProgressBarUtil.java
package org.example.util;

public class ProgressBarUtil {
    private static final String RESET = "\u001B[0m";
    private static final String GREEN = "\u001B[32m";


    public static void execute() throws InterruptedException {
        for (int i = 0; i < 100; i++) {
            printProgress(i, 100);
            Thread.sleep(250);
        }
        System.out.println();
    }

    private static void printProgress(int current, int total) {
        int progressWidth = 30;
        int progress = (int) ((double) current / total * progressWidth);
        int percent = (current * 100 / total);

        String time = String.format("[00:%02d]", current / 4);

        StringBuilder progressBar = new StringBuilder(time + " [");


        for (int i = 0; i < progressWidth; i++) {
            if (i < progress) {
                progressBar.append(GREEN).append("#");
            } else {
                progressBar.append(" ");
            }
        }
        progressBar.append(RESET).append("] ").append(percent).append("%");
        System.out.print("\r" + progressBar.toString());
    }


}

File: src/main/java/org/example/util/ProgressBarUtil.kt
package org.example.util

object ProgressBarUtil {
    private const val RESET = "\u001B[0m"
    private const val GREEN = "\u001B[32m"

    @JvmStatic
    fun printProgress(current: Int, total: Int) {
        val progressWidth = 30
        val progress = (current.toDouble() / total * progressWidth).toInt()
        val percent = (current * 100 / total)

        val progressBar = StringBuilder("[")

        for (i in 0..<progressWidth) {
            if (i < progress) {
                progressBar.append(GREEN).append("#")
            } else {
                progressBar.append(" ")
            }
        }
        progressBar.append(RESET).append("] ").append(percent).append("%")
        print("\r" + progressBar)
    }
}
